### ISSUE TITLE
Modal, Vanilla: AVT 2 issue: Keyboard users can't scroll static content when it overflows

### ISSUE THREAD

**Issue by elizabethsjudd** [carbon-design-system/carbon#3196]
## Environment

> Operating system

Mac

> Browser

All

> Automated testing tool and ruleset

n/a

> Assistive technology used to verify

Keyboard

## Detailed description

> What version of the Carbon Design System are you using?

v10.3.2

> What did you expect to happen?

I would be able to focus the content container so that I could use my arrow keys to scroll the content

> What happened instead?

I was not able to focus the container that had overflowing content so I was not able to read it

> What WCAG 2.1 checkpoint does the issue violate?

https://www.ibm.com/able/guidelines/ci162/keyboard.html

## Steps to reproduce the issue

Using only a keyboard

1. navigate within a modal that has overflowing content
2. notice you can never focus the content, making it so you can't scroll it unless you are using a mouse

> Please create a reduced test case in CodeSandbox

Includes the HTML fix:
https://codesandbox.io/embed/modal-overflow-content-srskh

## Additional information

* Need to add `tabindex=0` to the content container so that it is focusable.  This should be done only when the content overflows for the best experiences (dynamically via JS).
* Need input from @carbon-design-system/design around what the focus style for the content container should be

**Comment by elizabethsjudd** [comment]
@carbon-design-system/developers if you want to assign this to me I can take a look at it in the next couple of weeks

**Comment by elizabethsjudd** [comment]
@asudoh @dakahn In tooltips the practice was that it was just up to the developer to add the `tabindex` attribute themselves.  Because modals have a more dynamic size based on the screen size, is it okay if we test for this scenario in the JavaScript and apply it as needed (assume we'd also want to check on screen resize as well).

**Comment by asudoh** [comment]
> Need to add `tabindex=0` to the content container so that it is focusable. This should be done only when the content overflows for the best experiences (dynamically via JS).

We need to be careful about this part. Sounds that it requires layout query by JavaScript which is expensive.

**Comment by elizabethsjudd** [comment]
The options are to do it via JavaScript or to have it on their all the time which I've gotten a lot of push back from designers that it's an odd experience when it's not needed.  This isn't an optional feature because it's related to a11y.  We can debounce for the window resize.  I've personally never had an issue with this query on modals even on big data sites.

**Comment by dakahn** [comment]
@elizabethsjudd I'm personally cool with the _always on_ focus styles myself. Falls in line with our accessible-out-of-the-box ethos. Do you have an example of what this might look like?

**Comment by elizabethsjudd** [comment]
@dakahn this is the styling (which were just merged in to master) that was approved for design: https://github.com/carbon-design-system/carbon/issues/3261#issuecomment-507864039 and having it always on just means that if you are tabbing within a modal, you will always be able to highlight the content even if it's not scrollable.
